VTR_BUILD: unable to build with -DVTR_ENABLE_SANITIZE=ON

#### Current Behaviour
```
/usr/bin/ld.gold: error: cannot find -lasan
/usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/../../../../lib/libasan_preinit.o:asan_preinit.cc:__local_asan_preinit: error: undefined reference to '__asan_init'
/workspace/vtr-verilog-to-routing/libs/liblog/src/main.cpp:18: error: undefined reference to '__asan_init'
/workspace/vtr-verilog-to-routing/libs/liblog/src/main.cpp:18: error: undefined reference to '__asan_version_mismatch_check_v8'
/workspace/vtr-verilog-to-routing/libs/liblog/src/main.cpp:18: error: undefined reference to '__asan_register_globals'
CMakeFiles/test_log.dir/src/main.cpp.o:main.cpp:function _GLOBAL__sub_D_00099_0_main: error: undefined reference to '__asan_unregister_globals'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:38: error: undefined reference to '__asan_option_detect_stack_use_after_return'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:38: error: undefined reference to '__asan_stack_malloc_1'
/usr/include/bits/stdio.h:41: error: undefined reference to '__asan_report_load8'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:45: error: undefined reference to '__asan_option_detect_stack_use_after_return'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:45: error: undefined reference to '__asan_stack_malloc_1'
/usr/include/bits/stdio.h:41: error: undefined reference to '__asan_report_load8'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:62: error: undefined reference to '__asan_option_detect_stack_use_after_return'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:62: error: undefined reference to '__asan_stack_malloc_1'
/usr/include/bits/stdio.h:41: error: undefined reference to '__asan_report_load8'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:83: error: undefined reference to '__asan_option_detect_stack_use_after_return'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:83: error: undefined reference to '__asan_stack_malloc_1'
/workspace/vtr-verilog-to-routing/libs/liblog/src/log.cpp:92: error: undefined reference to '__asan_report_load8'
liblog.a(log.cpp.o):log.cpp:function _GLOBAL__sub_D_00099_0_log_stream: error: undefined reference to '__asan_unregister_globals'
liblog.a(log.cpp.o):log.cpp:function _GLOBAL__sub_I_00099_1_log_stream: error: undefined reference to '__asan_init'
liblog.a(log.cpp.o):log.cpp:function _GLOBAL__sub_I_00099_1_log_stream: error: undefined reference to '__asan_version_mismatch_check_v8'
liblog.a(log.cpp.o):log.cpp:function _GLOBAL__sub_I_00099_1_log_stream: error: undefined reference to '__asan_register_globals'
collect2: error: ld returned 1 exit status
make[3]: *** [libs/liblog/CMakeFiles/test_log.dir/build.make:85: libs/liblog/test_log] Error 1
make[2]: *** [CMakeFiles/Makefile2:596: libs/liblog/CMakeFiles/test_log.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[  3%] Built target libpugixml
-- VTR Version: 8.0.0-rc1+b2807fadc
Scanning dependencies of target libvtrutil
[  3%] Building CXX object libs/libvtrutil/CMakeFiles/libvtrutil.dir/vtr_version.cpp.o
[ 83%] Built target libabc
[ 83%] Linking CXX static library libvtrutil.a
[ 83%] Built target libvtrutil
make[1]: *** [Makefile:141: all] Error 2
make: *** [Makefile:76: all] Error 2
```

#### Steps to Reproduce
build with -DVTR_ENABLE_SANITIZE=ON

### Your Environment
#### uname -a
``` Linux jp-arch 5.2.1-arch1-1-ARCH #1 SMP PREEMPT Sun Jul 14 14:52:52 UTC 2019 x86_64 GNU/Linux ```

#### gcc -v 
``` Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release --enable-default-pie --enable-default-ssp --enable-cet=auto
Thread model: posix
gcc version 9.1.0 (GCC)```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -210,8 +210,7 @@ if(VTR_ENABLE_SANITIZE)
     #Enable sanitizers
     # -fuse-ld=gold force the gold linker to be used (required for sanitizers, but not enabled by default on some systems)
     set(SANITIZE_FLAGS "-g -fsanitize=address -fsanitize=leak -fsanitize=undefined -fuse-ld=gold")
-    message(STATUS "SANTIIZE_FLAGS: ${SANITIZE_FLAGS}")
-    link_libraries("-static-libasan") #Fixes 'ASan runtime does not come first in initial library list'
+    message(STATUS "SANITIZE_FLAGS: ${SANITIZE_FLAGS}")
 endif()
 
 #

